pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -1883,25 +1885,19 @@ func (service *SecurityPolicyService) buildRulePortsString(ports *[]v1alpha1.Sec
 			if idx == 0 {
 				portsString = portString
 			} else {
-				portsString = portsString + "-" + portString
+				portsString = portsString + "_" + portString
 			}
 		}
 	}
 	return util.GenerateTruncName(common.MaxNameLength, portsString, "", suffix, "", "")
 }
 
-func (service *SecurityPolicyService) BuildNetworkPolicyAllowPolicyName(uid string) string {
-	return fmt.Sprintf("%s-allow", uid)
-}
-
-func (service *SecurityPolicyService) BuildNetworkPolicyAllowPolicyID(uid string) string {
-	return fmt.Sprintf("%s_allow", uid)
-}
-
-func (service *SecurityPolicyService) BuildNetworkPolicyIsolationPolicyName(uid string) string {
-	return fmt.Sprintf("%s-isolation", uid)
+func (service *SecurityPolicyService) BuildNetworkPolicyAllowPolicyID(obj *networkingv1.NetworkPolicy) string {
+	suffix := "allow"
+	return util.GenerateIDByObjectWithSuffix(obj, suffix)
```

Comment:
This looks incorrect, it will generate the NetworkPolicy's ID as "$networkpolicy.name_$networkpolicy.uid_allow". If we use this as the id of the internal SecurityPolicy's UID, then the final generated NSX Policy ID should be "$networkpolicy.name_$networkpolicy.name_$networkpolicy.uid_allow" ?

Reply:
Thanks so much for pointing this wrong change. I reverted my change.